dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageProblemCauseAnnotator.java

```diff
@@ -47,7 +52,6 @@ public static void annotate(ClassPathResult rootResult, Iterable<LinkageProblem>
       ClassPathResult subtreeResult = cache.get(sourceArtifact);
       if (subtreeResult == null) {
         // Resolves the dependency graph with the source artifact at the root.
-        ClassPathBuilder classPathBuilder = new ClassPathBuilder();
         subtreeResult = classPathBuilder.resolveWithMaven(sourceArtifact);
```

Comment:
classPathBuilder looks like it can be null here.

Reply:
classPathBuilder parameter cannot be null; otherwise it throws NullPointerException here.

Initially I thought making the parameter `@Nullable`. However, the enforcer rule and LinkageCheckerMain both need to resolve dependency graphs using their ClassPathBuilder.

Reply:
It is a public method so anything can be passed. If you expect to be not null, it's worth checking.

Reply:
Added `checkNotNull()`s.